dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp

```diff
@@ -44,7 +44,7 @@ PageDirectoryPtr PageDirectoryFactory::createFromReader(String storage_name, WAL
 
     // After restoring from the disk, we need cleanup all invalid entries in memory, or it will
     // try to run GC again on some entries that are already marked as invalid in BlobStore.
-    dir->gcInMemEntries();
+    dir->gcInMemEntries(false);
```

Comment:
```suggestion
    dir->gcInMemEntries(/*return_removed_entries=*/false);
```
Add comment about why `false` is ok